### PR TITLE
blockformer

### PR DESCRIPTION
**Pull request by LeonWlw**
This PR is about implementation of blockformer in WeNet.
(Original paper: [https://arxiv.org/abs/2207.11697](https://arxiv.org/abs/2207.11697))
- Implementation Details
    - add se layer ensemble conformer encoder outputs
    - add se layer ensemble transformer decoder outputs
    - using relative positional encoding in decoder
In main branch, extracting features by torchaudio make a little worse results than the paper. I will push a branch using kaldi features for aishell recipe which can reproduce results in the paper.

### CODE REVIEW

**Comment by robin1001** [comment]
I think it's better if we add the experiment results on AIShell-1 and LibriSpeech, to show that we can get consistent and solid gain by using the model.

File: wenet/transformer/se_layer.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        self.fc = torch.nn.Sequential(
+            torch.nn.Linear(channel, channel // reduction, bias=False),
+            torch.nn.ReLU(inplace=True),
+            torch.nn.Linear(channel // reduction, channel, bias=False),
+            torch.nn.Sigmoid()
+        )
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        b, c, _, _ = x.size()
+        y = self.avg_pool(x).view(b, c)
```
Comment (kobenaxie) [inline wenet/transformer/se_layer.py:34]: avg_pool over T and D dim should consider pad_mask ?

**Comment by LeonWlw** [comment]
> I think it's better if we add the experiment results on AIShell-1 and LibriSpeech, to show that we can get consistent and solid gain by using the model.

@robin1001 results of aishell has been added

File: wenet/transformer/se_layer.py (hunk 1 above)
Comment (lezasantaizi) [reply wenet/transformer/se_layer.py:34]: thanks for your remind, we will update pad_mask to the code and retrain it .

**Comment by 903859154** [comment]
hi, i run blockformer in 3080 and it just used 30% - 40% gpu. I change batchsize bigger and numworker but it didn't work. So what should i do to take more use of gpu?
